core/input/input.cpp

```diff
@@ -92,13 +92,21 @@ void Input::set_mouse_mode(MouseMode p_mode) {
 		return;
 	}
 
+#ifndef WEB_ENABLED
 	// Allow to be set even if overridden, to see if the platform allows the mode.
+	// Web platform being an exception, due to the API for this being async there.
 	set_mouse_mode_func(p_mode);
 	mouse_mode = get_mouse_mode_func();
 
 	if (mouse_mode_override_enabled) {
 		set_mouse_mode_func(mouse_mode_override);
 	}
+#else
+	if (!mouse_mode_override_enabled) {
+		set_mouse_mode_func(p_mode);
+	}
+	mouse_mode = p_mode;
```

Comment:
I think this line can be removed, and the early return if `p_mode == mouse_mode` should be moved inside if `#ifndef WEB_ENABLED` block, and `Input::get_mouse_mode` should return `get_mouse_mode_func()` if `WEB_ENABLED` is defined (i.e. a more complete revert of the PR that caused the issue).

This fixes the issue in my game.

Reply:
The problem is that those changes would make the return value of `get_mouse_mode()` return the overridden value instead of the one set by the user.

Reply:
I created https://github.com/godotengine/godot/pull/101849 which goes more in depth about how `mouse_mode` is updated when the platform sets it async.